### PR TITLE
GetCovidData: retry for yesterday if there is no data for today

### LINKED ISSUE #49: getCovidData should try to return some data
The problem is when somebody is fetching the data before 3am as they might get an empty response. We should rather provide stale data than none at all.

when date is not explicitly provided in input, we should try to return data from `TODAY`. If data for `TODAY` is missing, try to get data for `YESTERDAY`, if those are missing too, return empty response.

when date is explicitly set in input, try to look them up and return empty response if they are missing. So no changes here.

Add `date` to response so the client can see which kind of data are they getting.

cc @Gh0stahCZ

### PR DESCRIPTION
**Pull request by marekaf**
closes #49

### CODE REVIEW

File: internal/functions/coviddata/get-covid-data.go [hunk 1]
```diff
@@ -92,13 +85,70 @@ func GetCovidData(w http.ResponseWriter, r *http.Request) error {
...
+		failed = true
 	}
 
-	increaseData := increase
+	if failed && !shouldRetry {
+		httputils.SendErrorResponse(w, r, rpccode.Code_INTERNAL, "Unknown error")
+		return
+	}
+
+	if failed && shouldRetry {
```
Comment (jendakol) [inline internal/functions/coviddata/get-covid-data.go]: I think this is more _fallback_ than _retry_, isn't it? :thinking:  
Anyway, I think these two blocks are so similar that they could be wrapped into sth like `func fetchData(date string) ((data, data), error)`.

**Review by jendakol — CHANGES_REQUESTED** [review]
It's certainly an improvement, however, I believe it can be improved even more :-)

File: internal/functions/coviddata/get-covid-data.go (hunk 1 above)
Comment (marekaf) [reply internal/functions/coviddata/get-covid-data.go]: right! I'll rename it to fallback, good point.

They are very similar, it is definitely boilerplate, though I'll have to come up with a better name, we already have fetchData in the same package :))

Comment (jendakol) [reply internal/functions/coviddata/get-covid-data.go]: `fetchAll`? Whatever :smile:

File: internal/functions/coviddata/get-covid-data.go [hunk 2]
```diff
@@ -52,41 +52,35 @@ func fetchTotals(ctx context.Context, client store.Client, date string) (*Totals
...
+	return &data, nil
 }
 
-func fetchIncrease(ctx context.Context, client store.Client, date string) (*IncreaseData, error) {
+func fetchCovidData(ctx context.Context, client store.Client, date string) (*TotalsData, *IncreaseData, error) {
 	logger := logging.FromContext(ctx)
 
-	snap, err := client.Doc(constants.CollectionCovidDataIncrease, date).Get(ctx)
-
+	//totalsData, err := fetchTotals(ctx, client, date)
```
Comment (jendakol) [inline internal/functions/coviddata/get-covid-data.go]: Delete? :-)

File: internal/functions/coviddata/get-covid-data.go [hunk 3]
```diff
@@ -52,41 +52,35 @@ func fetchTotals(ctx context.Context, client store.Client, date string) (*Totals
...
+		return nil, nil, err
 	}
 
-	logger.Infof("fetched firestore data: %+v", snap.Data())
-
-	var increase IncreaseData
-
-	if err := snap.DataTo(&increase); err != nil {
-		panic(fmt.Sprintf("could not parse input: %s", err))
+	//increaseData, err := fetchIncrease(ctx, client, date)
```
Comment (jendakol) [inline internal/functions/coviddata/get-covid-data.go]: Delete? :-)

**Review by jendakol — APPROVED** [review]

File: internal/functions/coviddata/get-covid-data.go (hunk 2 above)
Comment (marekaf) [reply internal/functions/coviddata/get-covid-data.go]: jj

File: internal/functions/coviddata/get-covid-data.go (hunk 1 above)
Comment (marekaf) [reply internal/functions/coviddata/get-covid-data.go]: done

**Pull request merged**